src/services/dashboards/dashboard-customize/modules/VariableSelectorDropdown.vue

```diff
@@ -123,38 +140,27 @@ const handleChangeVisible = () => {
     }
 };
 // TODO: search text binding
-const handleChangeContextMenuInput = (): void => {
-    // state.searchText = value;
+const handleChangeContextMenuInput = (text: string): void => {
+    state.searchText = text;
 };
 
-// reconvert to string | string[]
-watch(() => state.selected, (_selected) => {
-    let reconvertedSelected;
-    if (props.selectionType === 'SINGLE') {
-        reconvertedSelected = _selected[0].name;
-    } else reconvertedSelected = _selected.map((d) => d.name);
-    emit('change', reconvertedSelected);
-});
+const handleSelectOption = (_selected: MenuItem[]) => {
+    changeVariables(_selected);
+};
 
-onMounted(() => {
-    // convert to MenuItem for selected
-    if (props.defaultSelected) {
-        const defaultSelected = [] as MenuItem[];
-        if (Array.isArray(props.defaultSelected)) {
-            props.defaultSelected.forEach((d) => {
-                defaultSelected.push({ name: d, label: d });
-            });
-        } else defaultSelected.push({ name: props.defaultSelected, label: props.defaultSelected });
-        state.selected = defaultSelected;
+// helper
+const changeVariables = (changedSelected: MenuItem[]) => {
+    const variables = cloneDeep(dashboardDetailState.variables);
+    const reconvertedSelected = changedSelected.map((d) => d.name) as string[];
```

Comment:
```suggestion
    const reconvertedSelected: string[] = changedSelected.map((d) => d.name);
```
It looks like type assertion is not needed here